### ISSUE TITLE
Memory leaks

### ISSUE THREAD

**Issue by jack9267** [mikke89/RmlUi#263]
I have Visual Studio's leak detection enabled and it's going crazy when I use RmlUi.
Source\Core\ObserverPtr.cpp has a GetPool function that does a dirty quick "new".

```
static Pool< ObserverPtrBlock >& GetPool()
{
	// Wrap pool in a function to ensure it is initialized before use.
	// This pool must outlive all other global variables that derive from EnableObserverPtr. This even includes
	// user variables which we have no control over. For this reason, we intentionally let this leak.
	static Pool< ObserverPtrBlock >* pool =  new Pool< ObserverPtrBlock >(128, true);
	return *pool;
}
```

Can we please just fix this intentional leak? It makes it hard for me to debug my own leaks.

**Comment by mikke89** [comment]
Hm, I never had a problem here with valgrind. 

There isn't really anything to "fix" here, because this is intentional and necessary. I think the comment explains the situation pretty well. Is there no way you can add this to an ignore list or similar?

**Comment by jack9267** [comment]
![image](https://user-images.githubusercontent.com/7356277/149551408-6b1039e9-aeed-4eee-a47a-e608e63b52d8.png)
![image](https://user-images.githubusercontent.com/7356277/149551430-d360c2c2-9d3f-4cbb-94f4-bcd7a2a513a7.png)
![image](https://user-images.githubusercontent.com/7356277/149551450-dbebf2f6-98a8-477e-aff3-8fdffd9693cc.png)

The list goes on and on.

**Comment by mikke89** [comment]
Hm, the issue is that anything that eg. derives from `Rml::EventListener` then needs to outlive `Rml::Shutdown`, which will lead to unexpected crashes for many users, and more headaches than it solves.

In my opinion, it is perfectly reasonable for some (one-time) heap allocations to outlive the application, this to me sounds more like a limitation of the tool.

**Comment by jack9267** [comment]
The user should be deleting the Rml::EventListeners before calling Rml::ShutDown. Perhaps an optional bool to ShutDown or an additional function to shutdown the pool? I contain everything within a class so I don't keep anything from the library, to me it sounds like a user problem if they keep static globals. Alongs the ability to free the pool exists I'm happy. I need to be able to free up all the memory associated with the library for my needs. This is all for a multiplayer mod for the GTA series.

**Comment by jack9267** [comment]
Let me know though, I can make a pull request to even just add a function to free the observer pool.

**Comment by mikke89** [comment]
Is that the only one place causing issues?

Yeah, I'd be fine with a function to release the pool. Maybe next to the release textures and geometry functions? A PR would be welcome :)

**Comment by jack9267** [comment]
That sounds about right actually, now you mention the release textures and geometry functions. I'd still recommend freeing the pool if its empty in Rml::ShutDown (for those who don't hold the weird static references). Any naming preferences here for functions or the variable for the pool global?

Another alternative is to simply make the pool not a pointer (static) but I do prefer the function since I can call that when I need the memory freed.

**Comment by mikke89** [comment]
Great. Yes, actually, I agree that cleaning it up automatically if it's empty in Rml::Shutdown would be good. If we want a function in addition, maybe `ReleaseMemoryPool(s)`? But I'm open for suggestions.

**Comment by jack9267** [comment]
Just quickly wrote up a quick PR (not tested yet!)

https://github.com/mikke89/RmlUi/pull/265

**Comment by mikke89** [comment]
Fixed in #265, thanks!
